src/main/kotlin/org/pets/history/controller/PetController.kt

```diff
@@ -138,4 +139,29 @@ class PetController(private val petService: PetService) {
         @PathVariable petId: Long,
         @RequestBody @Valid medicalVisitIn: MedicalVisit
     ): MedicalVisit = petService.registerMedicalVisit(petId, medicalVisitIn)
+
+    @Operation(
+            summary = "Start a treatment",
+            description = "Start a treatment for a given pet id",
+    )
+    @ApiResponses(
+            value = [
+                ApiResponse(
+                        responseCode = "201",
+                        description = "Success",
+                        content = [
+                            Content(
+                                    mediaType = MediaType.APPLICATION_JSON_VALUE,
+                                    schema = Schema(implementation = Treatment::class),
+                            )
+                        ]
+                )
+            ]
+    )
+    @PostMapping("/{petId}/treatment")

```

Comment:
Suele ser una buena práctica tener el recurso en plural.

```suggestion
    @PostMapping("/{petId}/treatments")

```